dokan-migrator.php

```diff
@@ -0,0 +1,286 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ * **********************************************************************
+ */
+
+// don't call the file directly
+if ( ! defined( 'ABSPATH' ) ) {
+    exit;
+}
+
+/**
+ * Dokan_Migrator class
+ *
+ * @class Dokan_Migrator The class that holds the entire Dokan_Migrator plugin
+ *
+ * @since 1.0.0
+ */
+final class Dokan_Migrator {
+
+    /**
+     * Plugin version
+     *
+     * @var string
+     */
+    public $version = '1.0.0';
+
+    /**
+     * Dokan email classes.
+     *
+     * @var string
+     */
+    public static $email_class = [];
+
+    /**
+     * Dokan email templates.
+     *
+     * @var string
+     */
+    public static $template = [];
+
+    /**
+     * Dokan email actions.
+     *
+     * @var string
+     */
+    public static $actions = [];
+
+    /**
+     * Instance of self
+     *
+     * @var Dokan_Migrator
+     */
+    private static $instance = null;
+
+    /**
+     * Holds various class instances
+     *
+     * @since 1.0.0
+     *
+     * @var array
+     */
+    private $container = [];
+
+    /**
+     * Class constructor.
+     */
+    public function __construct() {
+        require_once __DIR__ . '/vendor/autoload.php';
+
+        // Define constants.
+        $this->define_constants();
+
+        // load the addon
+        add_action( 'dokan_loaded', array( $this, 'plugin_init' ) );
+    }
+
+    /**
+     * Initializes the WeDevs_Dokan() class
+     *
+     * Checks for an existing WeDevs_WeDevs_Dokan() instance
+     * and if it doesn't find one, creates it.
+     */
+    public static function init() {
+        if ( self::$instance === null ) {
+            self::$instance = new self();
+        }
+
+        return self::$instance;
+    }
+
+    /**
+     * Define all dokan migrator constant
+     *
+     * @since 1.0.0
+     *
+     * @return void
+     */
+    public function define_constants() {
+        define( 'DOKAN_MIGRATOR_PLUGIN_VERSION', $this->version );
+        define( 'DOKAN_MIGRATOR_FILE', __FILE__ );
+        define( 'DOKAN_MIGRATOR_DIR', dirname( DOKAN_MIGRATOR_FILE ) );
+        define( 'DOKAN_MIGRATOR_INC', DOKAN_MIGRATOR_DIR . '/includes' );
+        define( 'DOKAN_MIGRATOR_PLUGIN_ASSETS_DRI', DOKAN_MIGRATOR_DIR . '/assets' );
+        define( 'DOKAN_MIGRATOR_PLUGIN_ASSETS', plugins_url( 'assets', DOKAN_MIGRATOR_FILE ) );
+    }
+
+    /**
+     * Init the plugin.
+     *
+     * @return void
+     */
+    public function plugin_init() {
+        self::prevent_email_notification();
+        // Initialize the action hooks
+        $this->init_classes();
+
+        $this->show_admin_notice();
+    }
+
+    /**
+     * Show needed admin notice.
+     *
+     * @since 1.0.0
+     *
+     * @return void
+     */
+    public function show_admin_notice() {
```

Comment:
This method could be renamed as 'hooks' so that all necessary hooks for now and for future could be initiated inside the method.